optuna/_gp/acqf.py

```diff
@@ -117,6 +155,71 @@ def from_acqf_params(
         )
 
 
+@dataclass(frozen=True)
+class MultiObjectiveAcquisitionFunctionParams(AcquisitionFunctionParams):
+    acqf_params_for_objectives: list[AcquisitionFunctionParams]
+    non_dominated_box_lower_bounds: torch.Tensor
+    non_dominated_box_upper_bounds: torch.Tensor
+    fixed_samples: torch.Tensor
+
+    @classmethod
+    def from_acqf_params(
+        cls,
+        acqf_params_for_objectives: list[AcquisitionFunctionParams],
+        Y: np.ndarray,
+        n_qmc_samples: int,
+        qmc_seed: int | None,
+    ) -> MultiObjectiveAcquisitionFunctionParams:
+        def _get_non_dominated_box_bounds() -> tuple[torch.Tensor, torch.Tensor]:
+            loss_vals = -Y  # NOTE(nabenabe): Y is to be maximized, loss_vals is to be minimized.
+            pareto_sols = loss_vals[_is_pareto_front(loss_vals, assume_unique_lexsorted=False)]
+            ref_point = np.max(loss_vals, axis=0)
+            ref_point = np.nextafter(np.maximum(1.1 * ref_point, 0.9 * ref_point), np.inf)
+            lbs, ubs = get_non_dominated_box_bounds(pareto_sols, ref_point)
+            # NOTE(nabenabe): Flip back the sign to make them compatible with maximization.
+            return torch.from_numpy(-ubs), torch.from_numpy(-lbs)
+
+        fixed_samples = _sample_from_normal_sobol(
+            dim=Y.shape[-1], n_samples=n_qmc_samples, seed=qmc_seed
+        )
+        non_dominated_box_lower_bounds, non_dominated_box_upper_bounds = (
+            _get_non_dominated_box_bounds()
+        )
+        # The inverse_squared_lengthscales below is used only in optim_mixed.
+        # Since all the objectives are equally important, we simply use the mean of
+        # inverse_squared_lengthscales over all the objectives.
+        # cf. https://github.com/optuna/optuna/blob/v4.3.0/optuna/_gp/optim_mixed.py#L200-L209
+        inverse_squared_lengthscales = np.mean(
+            [
+                acqf_params.kernel_params.inverse_squared_lengthscales.detach().numpy()
+                for acqf_params in acqf_params_for_objectives
+            ],
+            axis=0,
+        )
```

Comment:
From what I can see in `_eval_ehvi`, the `kernel_params` in `MultiObjectiveAcquisitionFunctionParams` appears to be a dummy and is not actually used. Therefore, the `inverse_squared_lengthscales` does not need to hold any meaningful values either - would it make sense to simply set it to a dummy tensor, such as `torch.empty(len(params))`?

Reply:
It is a bit tricky because only `inverse_squared_lengthscales` is explicitly accessed externally; please see
- https://github.com/optuna/optuna/blob/v4.3.0/optuna/_gp/optim_mixed.py#L200-L209

So we cannot actually make this tensor a dummy version.

Reply:
I understood. Thanks for the explanation. BTW, how about using harmonic mean after taking square root? It ensures that the averaged lengthscale is used.

Reply:
Like this?

```python
        inverse_squared_lengthscales_array = np.array(
            [
                acqf_params.kernel_params.inverse_squared_lengthscales.detach().numpy()
                for acqf_params in acqf_params_for_objectives
            ]
        )
        mean_lengthscales = np.mean(1 / np.sqrt(inverse_squared_lengthscales_array))
        dummy_kernel_params = KernelParamsTensor(
            # inverse_squared_lengthscales is used in optim_mixed.py.
            inverse_squared_lengthscales=torch.from_numpy(1.0 / mean_lengthscales**2),
            # These parameters will not be used anywhere.
            kernel_scale=torch.empty(0),
            noise_var=torch.empty(0),
        )
```

Reply:
I applied your suggestion with the co-authorship:)